redisvl/extensions/session_manager/session.py

```diff
@@ -0,0 +1,293 @@
+                Defaults to None and will be replaced with the index name.
+            vectorizer Vectorizer: The vectorizer to create embeddings with.
+            distance_threshold float: The maximum semantic distance to be
+                included in the context. Defaults to 0.3
+            redis_client Optional[Redis]: A Redis client instance. Defaults to
+                None.
+
+
+        The proposed schema will support a single combined vector embedding
+        constructed from the prompt & response in a single string.
+
+        """
+        prefix = prefix or name
+        self._session_id = session_id
+        self._user_id = user_id
+        self._application_id = application_id
+        self._scope = scope
+
+        if vectorizer is None:
+            self._vectorizer = HFTextVectorizer(
+                model="sentence-transformers/all-mpnet-base-v2"
+            )
+
+        self.set_distance_threshold(distance_threshold)
+
+        schema = IndexSchema.from_dict({"index": {"name": name, "prefix": prefix}})
+
+        schema.add_fields(
+            [
+                {"name": "prompt", "type": "text"},
+                {"name": "response", "type": "text"},
+                {"name": "timestamp", "type": "numeric"},
+                {"name": "session_id", "type": "tag"},
+                {"name": "user_id", "type": "tag"},
+                {"name": "application_id", "type": "tag"},
+                {"name": "count", "type": "numeric"},
+                {"name": "token_count", "type": "numeric"},
+                {
+                    "name": "combined_vector_field",
+                    "type": "vector",
+                    "attrs": {
+                        "dims": self._vectorizer.dims,
+                        "datatype": "float32",
+                        "distance_metric": "cosine",
+                        "algorithm": "flat",
+                    },
+                 },
+            ]
+        )
+
+        self._index = SearchIndex(schema=schema)
+
+        if redis_client:
+            self._index.set_client(redis_client)
+            self._redis_client = redis_client
+        else:
+            self._index.connect(redis_url="redis://localhost:6379")
+            self._redis_client = Redis(decode_responses=True)
+
+        self._index.create(overwrite=False)
+
+        self._tag_filter = Tag("application_id") == self._application_id
+        if self._scope == 'user':
+            user_filter = Tag("user_id") == self._user_id
+            self._tag_filter = self._tag_filter & user_filter
+        if self._scope == 'session':
+            session_filter = Tag("session_id") == self._session_id
+            user_filter = Tag("user_id") == self._user_id
+            self._tag_filter = self._tag_filter & user_filter & session_filter
+
+
+    def clear(self):
+        """ Clears the chat session history. """
+        pass
+
+
+    def fetch_context(
```

Comment:
love it! I would push for `fetch_relevant` and `fetch_recent` for example to align the purposes of the interface and methods to end user (or something similar)